pkg/credentialprovider/gcp/metadata.go

```diff
@@ -298,3 +298,8 @@ func (g *containerRegistryProvider) Provide(image string) credentialprovider.Doc
 	}
 	return cfg
 }
+
+// truncateBlob is a sanitizing function
+func truncateBlob(blob []byte) []byte {
+	return blob[:16]
```

Comment:
This will panic if blob is small.

Reply:
Good catch!  Resolved by adopting a more opaque logging style overall.